src/main/java/io/github/genomicdatainfrastructure/daam/exceptions/ApplicationNotInCorrectStateException.java

```diff
@@ -0,0 +1,7 @@
+package io.github.genomicdatainfrastructure.daam.exceptions;
+
+public class ApplicationNotInCorrectStateException extends RuntimeException {
+    public ApplicationNotInCorrectStateException(String message) {
+        super(message);
```

Comment:
suggestion: to improve read,  the message is constant, you should move the text to this classe directly. And you should provide more meaningfull message, like the application id, the current application status and user id.